analysis_driver/dataset.py

```diff
@@ -344,6 +346,27 @@ def _find_pooling_step_for_artifact(art, expected_pooling_step_name=None, max_it
             )
         return art.input_artifact_list()
 
+    @property
+    def rapid_samples_by_lane(self):
+        if self._rapid_samples_by_lane is None:
+            self._rapid_samples_by_lane = {}
+
+            # TODO: move away from using the Lims API
```

Comment:
I believe this would be better placed in a GitHub issue than in a todo comment.

Reply:
@tcezard - what do we want to do here? This may require changes in the Rest API. The information we are currently getting from the Lims API consists of:
- lane number
- sample ID
- user sample ID
- project ID
- whether the sample is part of a pool (at the moment, determined by `len(reagent_labels) > 1`)
- number of samples in the lane (should always be 1)
- Rapid Analysis UDF

Reply:
I would split this in two:
 - The value we can get from the `lims/sample_info` endpoint
   - lane number
   - sample ID
   - project ID
   - user sample ID UDF, Rapid Analysis UDF (and other UDFs)
 - The rest can be obtained from the samplesheet (which still comes from the LIMS REST API)
   - whether the sample is part of a pool (at the moment, determined by len(reagent_labels) > 1)
   - number of samples in the lane (should always be 1)

Reply:
After discussion, there is a decision to be made - either of:
- take out all the Lims API queries (including advancing the Bioinformatics Processing step) and query the database instead (we'd need to add flowcell information to Reporting-App)
- keep the Lims API calls, and set up a dev instance of the Lims to be used in integration tests

The objective behind both of these is to stop patching the Lims in the integration tests. Removing Todo for now, pending further discussion.